test/sass/scss/scss_test.rb

```diff
@@ -4221,4 +4221,70 @@ def test_crlf
 SCSS
   end
 
+  def test_dynamic_include
+    assert_equal(<<CSS, render(<<SCSS))
+.foo {
+  invoked-with: 1px, 2px; }
+
+.foo {
+  invoked-with: passed as argument list; }
+CSS
+@mixin foo($args...) {
+  .foo { invoked-with: $args }
+}
+@include mixin(foo, 1px, 2px);
+$arglist: (foo, passed as argument list);
+@include mixin($arglist...);
+SCSS
+  end
+
+  def test_dynamic_includes_dynamic_include
+    assert_equal(<<CSS, render(<<SCSS))
+.foo {
+  invoked-with: 1px, 2px; }
+
+.foo {
+  invoked-with: passed as argument list; }
+CSS
+@mixin foo($args...) {
+  .foo { invoked-with: $args }
+}
+@include mixin(mixin, foo, 1px, 2px);
+$arglist: (mixin, foo, passed as argument list);
+@include mixin($arglist...);
+SCSS
+  end
+
+  def test_dynamic_include_prefers_local_definition
+    assert_equal(<<CSS, render(<<SCSS))
+.mixin {
+  dear-god-why: 2px; }
+CSS
+@mixin mixin($dear-god-why) {
+  .mixin { dear-god-why: $dear-god-why }
+}
+@include mixin(2px);
+SCSS
+  end
```

Comment:
I did this in the sass-spec tests actually. https://github.com/sass/sass-spec/blob/750725c5b4f4ac5d3656f92f0c0e99dbf7e9aa8c/spec/sass_3_5/dynamic-includes/simple/input.scss